include/oneapi/tbb/parallel_sort.h

```diff
@@ -218,13 +218,19 @@ void parallel_quick_sort( RandomAccessIterator begin, RandomAccessIterator end,
     See also requirements on \ref parallel_sort_iter_req "iterators for parallel_sort". **/
 //@{
 
+template<typename It>
+using iter_value_type = typename std::iterator_traits<It>::value_type;
+
 //! Sorts the data in [begin,end) using the given comparator
 /** The compare function object is used for all comparisons between elements during sorting.
     The compare object must define a bool operator() function.
     @ingroup algorithms **/
 template<typename RandomAccessIterator, typename Compare>
     __TBB_requires(std::random_access_iterator<RandomAccessIterator> &&
-                   compare<Compare, RandomAccessIterator>)
+                   compare<Compare, RandomAccessIterator> &&
+                   std::swappable<iter_value_type<RandomAccessIterator>> &&
+                   std::move_constructible<iter_value_type<RandomAccessIterator>> &&
```

Comment:
I suggest to simply use `std::movable` concept here. It will cover all three additional requirements. See here for details: https://en.cppreference.com/w/cpp/concepts/movable

Reply:
Goog idea!